### PR TITLE
Dependencies update (TS 1.8.7)

### PR DESCRIPTION
**Pull request by valorkin**
- all dependencies updated
- deps made immutable (save-exact)
- typescript added as dev dependency
- TS version set to 1.8.7

### CODE REVIEW

**Comment by nzakas** [comment]
Thanks. We follow the standard ESLint guidelines for pull requests:
http://eslint.org/docs/developer-guide/contributing/pull-requests

So, we need separate PRs for each commit and for the commit summaries to be in a specific format.

**Comment by valorkin** [comment]
@nzakas ready for merge

File: package.json [hunk 1]
```diff
@@ -26,18 +26,19 @@
     }
   ],
   "devDependencies": {
-    "chai": "^1.10.0",
```
Comment (nzakas) [inline package.json]: I'd prefer not to hardcode dependency versions.

File: package.json [hunk 2]
```diff
@@ -26,18 +26,19 @@
...
+    "eslint-config-eslint": "3.0.0",
+    "eslint-release": "0.3.0",
+    "istanbul": "0.4.2",
+    "leche": "2.1.1",
+    "mocha": "2.0.1",
+    "npm-license": "0.3.1",
+    "semver": "5.1.0",
+    "shelljs": "0.6.0",
+    "shelljs-nodecli": "0.1.1",
+    "typescript": "1.8.7"
```
Comment (nzakas) [inline package.json]: Why was this added? It's already in peerDependencies.

File: package.json (hunk 1 above)
Comment (valorkin) [inline package.json]: you prefer to have broken version because of someone changes?

File: package.json (hunk 2 above)
Comment (valorkin) [inline package.json]: using npm 3 it is not getting installed for development
and not changing npm module
so it is nice to have

File: package.json (hunk 1 above)
Comment (nzakas) [inline package.json]: Please revert these. Thanks.
